spyder/preferences/shortcuts.py

```diff
@@ -16,62 +16,94 @@
 from qtpy import PYQT5
 from qtpy.compat import from_qvariant, to_qvariant
 from qtpy.QtCore import (QAbstractTableModel, QModelIndex, QRegExp,
-                         QSortFilterProxyModel, Qt, Slot)
-from qtpy.QtGui import (QKeySequence, QRegExpValidator)
+                         QSortFilterProxyModel, Qt, Slot, QEvent)
+from qtpy.QtGui import (QKeySequence, QRegExpValidator, QIcon)
 from qtpy.QtWidgets import (QAbstractItemView, QApplication, QDialog,
-                            QDialogButtonBox, QGridLayout, QHBoxLayout, QLabel,
+                            QGridLayout, QHBoxLayout, QLabel,
                             QLineEdit, QMessageBox, QPushButton, QSpacerItem,
-                            QTableView, QVBoxLayout)
+                            QTableView, QVBoxLayout, QKeySequenceEdit)
 
 # Local imports
 from spyder.config.base import _, debug_print
+from spyder.config.main import CONF
 from spyder.config.gui import (get_shortcut, iter_shortcuts,
                                reset_shortcuts, set_shortcut)
 from spyder.preferences.configdialog import GeneralConfigPage
 from spyder.utils import icon_manager as ima
-from spyder.utils.qthelpers import get_std_icon
+from spyder.utils.qthelpers import get_std_icon, create_toolbutton
 from spyder.utils.stringmatching import get_search_scores, get_search_regex
 from spyder.widgets.helperwidgets import HTMLDelegate
 from spyder.widgets.helperwidgets import HelperToolButton
 
-
-MODIFIERS = {Qt.Key_Shift: Qt.SHIFT,
-             Qt.Key_Control: Qt.CTRL,
-             Qt.Key_Alt: Qt.ALT,
-             Qt.Key_Meta: Qt.META}
-
 # Valid shortcut keys
-SINGLE_KEYS = ["F{}".format(_i) for _i in range(1, 36)] + ["Delete", "Escape"]
-KEYSTRINGS = ["Tab", "Backtab", "Backspace", "Return", "Enter",

```

Comment:
Why did you remove all these KEYSTRINGS? Just curious, this was written by @goanpeca, not me.

Reply:
This is a good question.

Validating each `QKeyEvent.key()` in the `keyPressEvent` to a set of harcoded values was not working very well for international keyboards. For example,  doing `Ctrl+Alt+P` on my keyboard results in the `Qt.Key_Odiaeresis` key (Ö), which was obviously not in our list of predefined hardcoded keys. So, most of the `Ctrl+Alt` and `Ctrl+Alt+Shift` key sequences were not working with my keyboard with the previous Shortcut Editor.

So to solve that issue, I created the `ShortcutTranslator` that is used to convert `QKeyEvent` into `QKeySequence`. With this, pressing `Ctrl+Alt+Shift+P` results correctly in the key sequence `Ctrl+Alt+Shift+P` and not `Ctrl+Alt+Shift+Ö`.

In case an invalid key is still entered for some reason (due for example to the bug mentioned in https://github.com/spyder-ide/spyder/pull/7929#issuecomment-425202097), I check also that all characters of the key sequences are ASCII in `check_ascii`.

So basically, the harcoded KEYSTRINGS were not needed anymore, so that is why I removed them. That being said, it is quite possible that there is some corner cases that my new approach is missing. Worst scenario if this happens is that the new custom shortcut will simply do nothing.

Reply:
Ok, thanks a lot the detailed explanation!